sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctDocumentQueryExecutionContext.java

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.BadRequestException;
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.FeedResponse;
+import com.azure.cosmos.models.Resource;
+import reactor.core.publisher.Flux;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class DistinctDocumentQueryExecutionContext<T extends Resource> implements IDocumentQueryExecutionComponent<T> {
+    private final IDocumentQueryExecutionComponent<T> component;
+    private final DistinctMap distinctMap;
+    private final DistinctQueryType distinctQueryType;
+    private String lastHash;
+
+    public DistinctDocumentQueryExecutionContext(IDocumentQueryExecutionComponent<T> component,
+                                                 DistinctQueryType distinctQueryType,
+                                                 String previousHash) {
+        if (distinctQueryType == DistinctQueryType.None) {
+            throw new IllegalArgumentException("Invalid distinct query type");
+        }
+
+        if (component == null) {
+            throw new IllegalArgumentException("documentQueryExecutionComponent cannot be null");
+        }
+
+        this.distinctQueryType = distinctQueryType;
+        this.component = component;
+        this.distinctMap = DistinctMap.create(distinctQueryType, previousHash);
+    }
+
+    public static <T extends Resource> Flux<IDocumentQueryExecutionComponent<T>> createAsync(
+        Function<String, Flux<IDocumentQueryExecutionComponent<T>>> createSourceComponentFunction,
+        DistinctQueryType distinctQueryType,
+        String continuationToken) {
+
+        if (distinctQueryType == DistinctQueryType.Ordered) {
+            CosmosClientException dce = new BadRequestException("Ordered distinct queries are not supported yet");
+            return Flux.error(dce);
+        }
+
+        Utils.ValueHolder<DistinctContinuationToken> outDistinctcontinuationtoken = new Utils.ValueHolder<>();
+        DistinctContinuationToken distinctContinuationToken = new DistinctContinuationToken(null /*lasthash*/,
```

Comment:
This should be set in the else case of `if(continuationToken != null)`.

Reply:
Initialized it early as I am using outDistinctcontinuationtoken in if condition. And distinctContinuationToken is being used even if it doesnt go into else. So initialized these here